beacon_node/lighthouse_network/src/config.rs

```diff
@@ -347,3 +348,27 @@ pub fn gossipsub_config(network_load: u8, fork_context: Arc<ForkContext>) -> Gos
         .build()
         .expect("valid gossipsub configuration")
 }
+
+/// Helper function to determine if the IpAddr is a global address or not. The `is_global()`
+/// function is not yet stable on IpAddr.
+fn is_global(addr: &std::net::Ipv4Addr) -> bool {
```

Comment:
On merge, we should create an issue tracking https://github.com/rust-lang/rust/issues/27709